File: bootstrap/src/main/java/net/dontdrinkandroot/wicket/bootstrap/component/form/inputgroup/addon/InputGroupLabel.java
/*
 * Copyright (C) 2012-2017 Philip Washington Sorst <philip@sorst.net>
 * and individual contributors as indicated
 * by the @authors tag.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.addon;

import net.dontdrinkandroot.wicket.behavior.CssClassAppender;
import net.dontdrinkandroot.wicket.bootstrap.css.BootstrapCssClass;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.model.IModel;

import java.io.Serializable;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class InputGroupLabel extends Label
{
    public InputGroupLabel(String id)
    {
        super(id);
    }

    public InputGroupLabel(String id, Serializable label)
    {
        super(id, label);
    }

    public InputGroupLabel(String id, IModel<?> model)
    {
        super(id, model);
    }

    @Override
    protected void onInitialize()
    {
        super.onInitialize();
        this.add(new CssClassAppender(BootstrapCssClass.INPUT_GROUP_ADDON));
    }
}

File: bootstrap/src/main/kotlin/net/dontdrinkandroot/wicket/bootstrap/component/form/inputgroup/addon/InputGroupLabel.kt
package net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.addon

import net.dontdrinkandroot.wicket.behavior.CssClassAppender
import net.dontdrinkandroot.wicket.bootstrap.css.BootstrapCssClass
import org.apache.wicket.markup.html.basic.Label
import org.apache.wicket.model.IModel

class InputGroupLabel(id: String, model: IModel<*>? = null) : Label(id, model) {

    override fun onInitialize() {
        super.onInitialize()
        add(CssClassAppender(BootstrapCssClass.INPUT_GROUP_ADDON))
    }
}